src/index.js

```diff
@@ -115,200 +115,208 @@ class UglifyJsPlugin {
   apply(compiler) {
     const requestShortener = new RequestShortener(compiler.context);
 
-    compiler.plugin('compilation', (compilation) => {
+    compiler.hooks.compilation.tap('UglifyJsWebpackPlugin', (compilation) => {
```

Comment:
Is `UglifyJsWebpackPlugin` accessible as a hook or is this just the plugin name for e.g interception ? `UglifyJsPlugin` ?  What's the naming convention here ? :)

Reply:
@michael-ciniawsky - Check the tapable changes for 4.x. This is something you are going to see a lot of on the contrib side for Webpack 4

Reply:
Yep, that's why I'm asking :)

Reply:
This is the plugin name! :-D